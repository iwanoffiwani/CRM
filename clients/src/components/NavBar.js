import React from 'react'
import { NavLink } from 'react-router-dom'
import { makeStyles } from '@material-ui/core/styles'
import List from '@material-ui/core/List'
import ListItem from '@material-ui/core/ListItem'
import ListItemIcon from '@material-ui/core/ListItemIcon'
import ListItemText from '@material-ui/core/ListItemText'
import FormatListNumberedIcon from '@material-ui/icons/FormatListNumbered'
import SettingsIcon from '@material-ui/icons/Settings'

const useStyles = makeStyles(theme => ({
  link: {
    display: 'flex',
    alignItems: 'center',
    textDecoration: 'none',
    color: 'inherit'
  }
}))

const NavBar = () => { 

  const classes = useStyles()

  return ( 
    <nav className='navbar'>
      <List>
        <ListItem>
          <NavLink to='/' className={classes.link}>
            <ListItemIcon>
              <FormatListNumberedIcon />
            </ListItemIcon>
            <ListItemText primary='Заявки' />
          </NavLink>
        </ListItem>
        <ListItem>
          <NavLink to='/settings' className={classes.link}>
            <ListItemIcon>
              <SettingsIcon />
            </ListItemIcon>
            <ListItemText primary='Настройки' />
          </NavLink>
        </ListItem>
      </List>
    </nav>
  )
}

export default NavBar