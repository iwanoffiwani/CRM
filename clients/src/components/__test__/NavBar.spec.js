import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import NavBar from '../NavBar'

describe(`<NavBar />`, () => {
  it(`+++ renderer component`, () => {
    const wrapper = shallow(<NavBar />)
    return expect(toJson(wrapper)).toMatchSnapshot()
  })
})