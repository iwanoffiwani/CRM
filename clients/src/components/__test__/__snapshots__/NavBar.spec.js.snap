// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> +++ renderer component 1`] = `
<nav
  className="navbar"
>
  <WithStyles(ForwardRef(List))>
    <WithStyles(ForwardRef(ListItem))>
      <NavLink
        className="makeStyles-link-1"
        to="/"
      >
        <WithStyles(ForwardRef(ListItemIcon))>
          <FormatListNumberedIcon />
        </WithStyles(ForwardRef(ListItemIcon))>
        <WithStyles(ForwardRef(ListItemText))
          primary="Заявки"
        />
      </NavLink>
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))>
      <NavLink
        className="makeStyles-link-1"
        to="/settings"
      >
        <WithStyles(ForwardRef(ListItemIcon))>
          <SettingsIcon />
        </WithStyles(ForwardRef(ListItemIcon))>
        <WithStyles(ForwardRef(ListItemText))
          primary="Настройки"
        />
      </NavLink>
    </WithStyles(ForwardRef(ListItem))>
  </WithStyles(ForwardRef(List))>
</nav>
`;
